### PR TITLE
#9: Handle nil query params and allow for various query types

### PR DESCRIPTION
**Pull request by sh-ura**

### CODE REVIEW

**Comment by sh-ura** [comment]
@ljdursi I am not sure why the Travis builds are failing, it builds fine locally and `operations.NewMainGetVariantsUnprocessableEntity()` seems to be correctly defined from what I can see. Could you check to make sure that imports/dependencies for Travis builds are being pulled from the Pull Request that is being built, rather than paths to master?

**Comment by sh-ura** [comment]
PR #22 is based on this branch and seems to be passing Travis checks fine...

**Review by ljdursi — APPROVED** [review]

File: variant-service/api/restapi/configure_variant_service.go [hunk 1]
```diff
@@ -110,10 +120,35 @@ func configureAPI(api *operations.VariantServiceAPI) http.Handler {
...
 		}
 
-		query := tx.Where(fmt.Sprintf("chromosome = '%s' AND start BETWEEN %d AND %d",
-			*params.Chromosome, *params.Start, *params.End))
+		conditions := ""
+
+		if params.Chromosome != nil {
+			conditions = fmt.Sprintf(addAND(conditions) + "chromosome = '%s'", *params.Chromosome)
+		} else {
+			// TODO do we want to do this? If so, add 422001 to error code doc. If not, remove here and in swagger.yml
```
Comment (ljdursi) [inline variant-service/api/restapi/configure_variant_service.go]: I don't think we need to do this; if the query isn't clearly invalid or likely to be harmful, I don't think we need to write code to forbid any particular search combination.

**Comment by sh-ura** [comment]
@ljdursi Also suggested the following:

> I don’t think it’s worth writing code to disallow silly queries, unless they’re clearly invalid or potentially harmful.  (Listing all variants known in the database would probably fall into the latter, but for these purposes it’s probably fine)

Since such a query would cause massive slowdowns in a production service, and would likely only be requested in error or in malice, I will disallow the "get all variants" query prior to merging this PR.

**Comment by sh-ura** [comment]
Travis tests succeeded and completed following commit ["#9: Permit all non-harmful valid query parameter combinations"](https://github.com/CanDIG/go-model-service/pull/21/commits/7f36346f687abc4a7852919563e64558e94eac41), but are now failing seemingly due to the addition of the "operations.NewMainGetVariantsForbidden" function in the most recent commit.

It seems that new Go-Swagger-generated functions trip up Travis tests in the commit in which they are introduced, but tests for subsequent commits succeed. 

Regardless, the PR has been locally built and manually tested so it will be merged.

**Pull request merged**